woocommerce/woocommerce
# Add generic error boundary component

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Related to https://github.com/woocommerce/woocommerce/issues/32284

This PR adds an experimental generic error boundary component to the @woocommerce/components package. This component can be used to catch errors in the component tree and display a fallback UI.

The UI is based on the design suggested in the issue, although it might a bit outdated but should be a good starting point.

I also fixed the storybook build issue. It was failing because some dependencies are using react 18 and one package failed to find react/runtime.

![Screenshot 2024-06-11 at 18 09 46](https://github.com/woocommerce/woocommerce/assets/4344253/6269bc90-915d-4fed-a0c6-4234b7761b6a)




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. cd ./tools/storybook
2. Run `pnpm storybook` to start the storybook server.
3. Go to http://localhost:6007/?path=/story/woocommerce-admin-experimental-error-boundary
4. Play with the component and see if it works as expected.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
